### PR TITLE
Update developer guide for add&delete position. 

### PR DESCRIPTION
**Pull request by Harry-Gao-H**
Update developer guide for add&delete position. 
Add to NFR.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -272,14 +272,53 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+    * 1a1. MrTechRecruiter shows an error message.
+    
+    Use case ends.
+    
+
+**Use case: Delete a new position**
+
+**MSS**
+
+1. User requests to list positions
```
Comment (BenjaminTan99) [inline docs/DeveloperGuide.md:306]: I think this listing itself can be another use case (e.g. Request list of positions) such that the line becomes something like
1. User requests and is shown all current list positions (UC3).
2. User requests to delete...

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -272,14 +272,53 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+4. MrTechRecruiter deletes the position
+    
+    Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+    Use case ends.
+* 3a The given name of the position is invalid.
+  * 3a1. MrTechRecruiter shows an error message.
```
Comment (BenjaminTan99) [inline docs/DeveloperGuide.md:318]: Similar comments as the earlier error message comment.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -293,7 +332,7 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 **Extensions**
 
 * 1a. The format is invalid.
-    * 3a1. MrTechRecruiter shows an error message.
+    * 1a1. MrTechRecruiter shows an error message.
```
Comment (BenjaminTan99) [inline docs/DeveloperGuide.md:335]: Similar to the earlier error message comments.

**Review by BenjaminTan99 — CHANGES_REQUESTED** [review]

**Review by BenjaminTan99 — CHANGES_REQUESTED** [review]
Needs some adjustments for user stories.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -272,14 +272,53 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
+
 *{More to be added}*
 
 ### Use cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is the `MrTechRecruiter` and the **Actor** is the `user`, unless specified otherwise)
+
+**Use case: Add a new position**
```
[the remark this answers is not in the document]
Comment (Harry-Gao-H) [reply docs/DeveloperGuide.md:287]: That's a good idea I think. Will add indices to **Use Case**

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -272,14 +272,53 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+**MSS**
+
+1. User requests to add a new position, with necessary details
+2. MrTechRecruiter adds the new position and shows a success message
+
+   Use case ends.
+
+* 1a. The format is invalid.
```
[the remark this answers is not in the document]
Comment (Harry-Gao-H) [reply docs/DeveloperGuide.md:296]: Use case detects the **interaction** between the user and the system, so I think 1a in your comment may be unnecessary. 
And `requests for the correct data` can be a part of `show an error message`?

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Harry-Gao-H) [reply docs/DeveloperGuide.md:306]: `list position` can be another use case that later we can refer to I think.

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -272,14 +272,53 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+1. User requests to list positions
+2. MrTechRecruiter shows a list of positions
+3. User requests to delete a specific position in the list
+4. MrTechRecruiter deletes the position
+    
+    Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
```
[the remark this answers is not in the document]
Comment (Harry-Gao-H) [reply docs/DeveloperGuide.md:315]: Thanks for pointing that out :)

File: docs/DeveloperGuide.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (BenjaminTan99) [reply docs/DeveloperGuide.md:287]: ok i understand! no worries

File: docs/DeveloperGuide.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (BenjaminTan99) [reply docs/DeveloperGuide.md:296]: Yes possible.

**Review by BenjaminTan99 — APPROVED** [review]

**Pull request merged**